src/utilities/sync.js

```
@@ -65,10 +74,149 @@ altspace.utilities.sync = (function() {
+     * Returns a promise that will fufill with a [Session]{@link module:altspace/utilities/sync~Session}.
+     *
+     * @method connect
+     * @param {Object} config
+     * @param {String} config.authorId A unique identifier for yourself or your organization
+     * @param {String} config.appId The name of your app
+     * @param {String} [config.baseRefUrl] Override the base reference. Set this to use your own Firebase.
+     * @param {String} [config.instanceId] Override the instanceId. Can also be overriden using a query parameter.
+     * @param {String} [config.spaceId] Override the spaceId. Can also be overriden using a query parameter.
+     * @param {String} [config.userId] Override the userId. Can also be overriden using a query parameter.
+     * @return {Promise}
+     * @memberof module:altspace/utilities/sync
+     **/
+    //todo clients
+    function connect(config) {
+        config = config || {};
+
+        var url = new Url();
+
+        // Our ref used for example apps. Data may be cleared periodically.
+        var baseRefUrl = config.baseRefUrl || 'https://altspace-apps.firebaseio.com/apps/examples/';
+        var baseRef = new Firebase(baseRefUrl);
+
+        // Gather query paramaters (some may only be used as testing overrides)
+        var instanceId = config.instanceId || url.query['altspace-sync-instance'];
+        var spaceId = config.spaceId || url.query['altspace-sync-space'];
+        var userId = config.userId || url.query['altspace-sync-user'];
+
+        var tasks = [authenticate(baseRef)];
+        if (inAltspace) {
+            if (!spaceId) tasks.unshift(altspace.getSpace());
+            if (!userId) tasks.unshift(altspace.getUser());
+        }
+
+        function getRefs() {
+            var refs = {};
+
+            var projectId = getProjectId(config.appId, config.authorId, canonicalUrl);
```

Looks like we should throw an error if appId or authorId are excluded since they form the key in Firebase. IIRC, we only used canonicalUrl for CodePen.